packages/compiler/server/server-utils.ts

```diff
@@ -0,0 +1,54 @@
+import { isWhiteSpace } from "../core/charcode.js";
+import { Comment, TypeSpecScriptNode } from "../core/index.js";
+
+/**
+ * Find the comment that is at given position, if any.
+ *
+ * A comment is at the given position if {@link Comment.pos} <= position <
+ * {@link Comment.end}. Unlike {@link getNodeAtPosition}, the end node is
+ * not included since comments can be adjacent to each other with no trivia
+ * or punctuation between them.
+ *
+ * @internal
+ */
+export function getCommentAtPosition(script: TypeSpecScriptNode, pos: number): Comment | undefined {
+  // Comments are ordered by increasing position, use binary search
+  let low = 0;
+  let high = script.comments.length - 1;
```

Comment:
is there any issues to the behavior if we choose not to parse the comments?

Reply:
Yes, it will break. Good point. Maybe this should throw. It's only in server-utils now so not a problem, but I could see this becoming useful elsewhere and going public. So I'll put in a throw now.

Reply:
Done.